### DISCUSSION TITLE
cannot access `C:\WINDOWS\System32\OpenSSH`

### DISCUSSION THREAD

**Discussion by Zoybean** [nushell/nushell#11754] [Q&A] [answered]
I don't know what's going on with my setup, but from git bash and from windows explorer, I can access the `C:\WINDOWS\System32\OpenSSH` folder, but `nu` cannot access it.
Demonstration:
```
> bash
> cd /c/WINDOWS/System32/OpenSSH
> nu
> pwd
C:\WINDOWS\System32\OpenSSH
> cd (pwd)
Error: nu::shell::directory_not_found

  × Directory not found
   ╭─[entry #2:1:1]
 1 │ cd (pwd)
   ·    ──┬──
   ·      ╰── directory not found
   ╰────
  help: C:\WINDOWS\System32\OpenSSH does not exist
```

The closest I've found is this reddit post [Can't access "C:\Windows\System32\OpenSSH\ssh.exe" from C# application](https://www.reddit.com/r/csharp/comments/i06zdq/cant_access_cwindowssystem32opensshsshexe_from_c/) which seems to match my symptoms, but it says it happens for 32-bit applications. I built nushell from source - would it be 32-bit by default?

If that's not it, my best guess is that there's some permissions set on the directory that allow access only to some programs and not others, but I'm not too familiar with Windows' permission system.

I've also had a look at [Git under nushell not using added ssh-key](https://stackoverflow.com/questions/77856195/git-under-nushell-not-using-added-ssh-key) but in my case it's not a git issue, it's all of `ssh` being missing.

Also note, this is on my work computer, and I haven't had this problem on my home PC.

**Comment by fdncred** [comment]
I think this is a permissions thing. System32 is a protected folder. I don't have issues accessing it but I'm running from an admin account.

**Comment by ChrisDenton** [comment]
> I built nushell from source - would it be 32-bit by default?

What does `rustup show` display?

**Reply by Zoybean** [reply]
I forgot to check at work, and at home I can access SYSTEM32.
within the nushell repo on my home machine, `rustup show` prints (with some paths redacted):
```
Default host: x86_64-pc-windows-msvc
rustup home:  C:\...\.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-msvc (default)
nightly-x86_64-pc-windows-msvc
1.61.0-x86_64-pc-windows-msvc
1.65.0-x86_64-pc-windows-msvc
1.66.1-x86_64-pc-windows-msvc
1.67.1-x86_64-pc-windows-msvc
1.72.1-x86_64-pc-windows-msvc
1.74.1-x86_64-pc-windows-msvc

active toolchain
----------------

1.74.1-x86_64-pc-windows-msvc (overridden by 'C:\...\nushell\rust-toolchain.toml')
rustc 1.74.1 (a28077b28 2023-12-04)
```

I'll get back with the output at work, on monday.

**Reply by Zoybean** [reply, marked as answer]
The output of `rustup show` at work.
```
Default host: i686-pc-windows-msvc
rustup home:  C:\...\.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-gnu
stable-x86_64-pc-windows-msvc (default)
nightly-i686-pc-windows-msvc
nightly-x86_64-pc-windows-msvc
1.65.0-i686-pc-windows-msvc
1.67.1-i686-pc-windows-msvc
1.70.0-i686-pc-windows-msvc
1.72.1-i686-pc-windows-msvc
1.73.0-i686-pc-windows-msvc
1.74.1-i686-pc-windows-msvc

active toolchain
----------------

1.74.1-i686-pc-windows-msvc (overridden by 'C:\...\nushell\rust-toolchain.toml')
rustc 1.74.1 (a28077b28 2023-12-04)
```

The difference between these outputs (specifically the default toolchain, active toolchain, and default host) tells me it might have been a rustup configuration issue (my default host was `i686` instead of `x86_64`). After changing the setting in my rustup `settings.toml` and rebuilding, nushell can now access `C:\WINDOWS\System32\OpenSSH`
